application-licensing-licensor/application-licensing-licensor-api/src/main/java/com/xwiki/licensing/internal/GetTrialLicenseListener.java

```diff
@@ -95,13 +106,17 @@ public void tryGenerateTrialLicenseRecursive(ExtensionId extensionId)
             trialLicenseGenerator.generateTrialLicense(extensionId);
         } else {
             InstalledExtension installedExtension = installedExtensionRepository.getInstalledExtension(extensionId);
+
             if (installedExtension != null) {
                 Collection<ExtensionDependency> dependencies = installedExtension.getDependencies();
-
                 for (ExtensionDependency dependency : dependencies) {
-                    ExtensionId dependencyId =
-                        new ExtensionId(dependency.getId(), dependency.getVersionConstraint().getVersion());
-                    tryGenerateTrialLicenseRecursive(dependencyId);
+                    try {
+                        Extension extension = repositoryManager.resolve(dependency);
```

Comment:
I think you should use ``installedExtensionRepository`` to resolve the dependency for two reasons:

* we want to know which **installed** extension satisfies that dependency
* we're not interested in **core** licensed extensions so we should only look in the installed extension repository